### PR TITLE
Rsvp resubmission

### LINKED ISSUE #27: RSVP resubmission
When resubmitting from the RSVP page, modify the guest's data in the database, specifically when submitting using the pokemon code link or while they are logged in.

### PR DESCRIPTION
**Pull request by lvoytek**
A user can now resubmit their RSVP with new info, including new associates, etc. Closes #27

### CODE REVIEW

File: server/src/associate/associate.service.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ export class AssociateService {
 			},
 		});
 
-		return check1 !== undefined || check2 !== undefined;
+		return !!check1 || !!check2;
```
Comment (lboje) [inline server/src/associate/associate.service.ts:62]: I feel like there has to be a better name for these variables. Idk what though

File: client/src/app/components/forms/rsvp/rsvp.component.ts [hunk 2]
```diff
@@ -151,7 +164,7 @@ export class RsvpFormComponent implements OnChanges {
 
 			guestFormGroup.controls['firstName'].setValue(firstName);
 			guestFormGroup.controls['lastName'].setValue(lastName);
-			guestFormGroup.controls['attending'].setValue(isGoing);
+			guestFormGroup.controls['attending']?.setValue(isGoing);
```
Comment (lboje) [inline client/src/app/components/forms/rsvp/rsvp.component.ts:167]: why optional?

File: server/src/contact/contact.service.ts [hunk 3]
```diff
@@ -54,14 +54,34 @@ export class ContactService {
...
+	 * @param contact The new or updated contact info
+	 * @return The resulting contact data
+	 */
+	async createOrUpdate(
+		guest: primaryData,
+		contact: contactData,
+	): Promise<contactData> {
+		try {
+			return await this.create(guest, contact);
+		} catch (err) {
```
Comment (lboje) [inline server/src/contact/contact.service.ts:69]: is there a specific error that’s thrown back by typeorm here if specifically the resource already exists? like a 403 or 409 or something?

File: server/src/rsvp/rsvp.controller.ts [hunk 4]
```diff
@@ -147,10 +164,15 @@ export class RsvpController {
 					);
 				}
 			}
-			rsvpInfo.associates = associates;
 		}
 
-		return { ...guest, ...contactInfo, ...rsvpInfo };
+		return {
+			primary: !!guest,
```
Comment (lboje) [inline server/src/rsvp/rsvp.controller.ts:170]: :/ why

File: server/src/rsvp/rsvp.service.ts [hunk 5]
```diff
@@ -53,13 +54,35 @@ export class RsvpService {
 		return null;
 	}
 
+	/**
+	 * Attempt to create a new RSVP data entry, if it already exists just update it
+	 * @param guest The primaryData of the guest to add RSVP info for
+	 * @param rsvp The new or updated RSVP info
+	 * @return The resulting RSVP data
+	 */
+	async createOrUpdate(
```
Comment (lboje) [inline server/src/rsvp/rsvp.service.ts:63]: did you implement this twice?

Comment (lboje) [reply server/src/rsvp/rsvp.service.ts:63]: nvm I think it’s slightly different. Same question as above, then

File: client/src/app/components/forms/rsvp/rsvp.component.ts (hunk 2 above)
Comment (lvoytek) [reply client/src/app/components/forms/rsvp/rsvp.component.ts:167]: The associates don't have this item so having it be mandatory breaks autofill

Comment (lboje) [reply client/src/app/components/forms/rsvp/rsvp.component.ts:167]: don’t we need this to be mandatory, though?

Comment (lboje) [reply client/src/app/components/forms/rsvp/rsvp.component.ts:167]: so they can’t just submit without saying if they’re going

File: server/src/contact/contact.service.ts (hunk 3 above)
Comment (lvoytek) [reply server/src/contact/contact.service.ts:69]: It returns QueryFailedError. I can check for that specifically if you want

File: server/src/rsvp/rsvp.controller.ts (hunk 4 above)
Comment (lvoytek) [reply server/src/rsvp/rsvp.controller.ts:170]: idk, not sure what to do with the return yet but I can send back everything or nothing if you want

File: server/src/contact/contact.service.ts (hunk 3 above)
Comment (lboje) [reply server/src/contact/contact.service.ts:69]: if they don’t further specify that it’s due to the resource existing already, it’s not much help to check.

File: server/src/rsvp/rsvp.service.ts (hunk 5 above)
Comment (lvoytek) [reply server/src/rsvp/rsvp.service.ts:63]: No but I created the new function in both rsvp.service and contact.service

File: server/src/rsvp/rsvp.controller.ts (hunk 4 above)
Comment (lboje) [reply server/src/rsvp/rsvp.controller.ts:170]: but why are we doing booleans? I literally don’t understand sorry

File: client/src/app/components/forms/rsvp/rsvp.component.ts (hunk 2 above)
Comment (lvoytek) [reply client/src/app/components/forms/rsvp/rsvp.component.ts:167]: It is mandatory for the main user, this is just autofill

Comment (lboje) [reply client/src/app/components/forms/rsvp/rsvp.component.ts:167]: k

File: server/src/contact/contact.service.ts (hunk 3 above)
Comment (lvoytek) [reply server/src/contact/contact.service.ts:69]: fair

File: server/src/associate/associate.service.ts (hunk 1 above)
Comment (lvoytek) [reply server/src/associate/associate.service.ts:62]: I can make that clearer

File: server/src/rsvp/rsvp.controller.ts (hunk 4 above)
Comment (lvoytek) [reply server/src/rsvp/rsvp.controller.ts]: Returning everything again

**Review by lboje — APPROVED** [review]

**Pull request merged**